finesse/gui/event_counter.py

```diff
@@ -40,16 +42,20 @@ def __init__(
         # Subscribe to devices' open/close messages
         for name in device_names:
             pub.subscribe(self.increment, f"device.opened.{name}")
-            pub.subscribe(self.decrement, f"device.closed.{name}")
+            pub.subscribe(self._on_device_closed, f"device.closed.{name}")
 
     def increment(self) -> None:
         """Increase the counter by one and run callback if target reached."""
         self._count += 1
         if self._count == self._target_count:
             self._on_target_reached()
 
-    def decrement(self, **kwargs) -> None:
+    def decrement(self) -> None:
         """Decrease the counter by one and run callback if count drops below target."""
         self._count -= 1
         if self._count == self._target_count - 1:
             self._on_below_target()
+
+    def _on_device_closed(self, instance: DeviceInstanceRef) -> None:
```

Comment:
I seem to remember we had removed this before as it seemed redundant with `decrement` (`instance` is not used here). Why is it back?

Reply:
> "because pypubsub is not happy, otherwise"

This ^^ :wink:

As you suggested before, you can specify all your pubsub topics + args in .py files and that might also work. That's a job for another day though.